handlers.go

```diff
@@ -3600,6 +3604,67 @@ func (s *Server) handleDeleteCollection(c echo.Context, u *User) error {
 	return c.NoContent(http.StatusOK)
 }
 
+// handleDeleteContentFromCollection godoc
+// @Summary      Deletes a content from a collection
+// @Description  This endpoint is used to delete an existing content from an existing collection. If two or more files with the same contentid exist in the collection, delete the one in the specified path
+// @Tags         collections
+// @Param        coluuid path string true "Collection ID"
+// @Param        contentid path string true "Content ID"
+// @Param        path query string false "Path in which delete files (if found)"
+// @Produce      json
+// @Success      200  {object}  string
+// @Failure      400  {object}  util.HttpError
+// @Router       /collections/{coluuid}/content/{contentid} [delete]
+func (s *Server) handleDeleteContentFromCollection(c echo.Context, u *util.User) error {
+	coluuid := c.Param("coluuid")
+	contentid := c.Param("contentid")
+	path := c.QueryParam("path")
+
+	col, err := collections.GetCollection(coluuid, s.DB, u)
+	if err != nil {
+		return err
+	}
+
+	content, err := util.GetContent(contentid, s.DB, u)
+	if err != nil {
+		return err
+	}
+
+	refs := []collections.CollectionRef{}
+	if err := s.DB.Model(collections.CollectionRef{}).
+		Where("collection = ?", col.ID).
+		Where("content = ?", content.ID).
+		Scan(&refs).Error; err != nil {
+		return err
+	}
+
+	if len(refs) > 1 {
+		// file is present more than once in collection, need path to figure out which exactly to delete
+		if path == "" {
+			return &util.HttpError{
+				Code:    http.StatusNotFound,
+				Reason:  util.ERR_PATH_NOT_SPECIFIED,
+				Details: fmt.Sprintf("could not determine file to delte, found %d with specified contentID but path was not specified", len(refs)),
+			}
+		}
+
+		// filter again, now with specified path
+		if err := s.DB.Model(collections.CollectionRef{}).
+			Where("collection = ?", col.ID).
+			Where("content = ?", content.ID).
+			Where("path = ?", path).
+			Scan(&refs).Error; err != nil {
+			return err
+		}
+
+	}
+
+	if err := s.DB.Delete(&refs).Error; err != nil {
```

Comment:
should this not be done in `if len(refs) > 1 {` scope?

Reply:
The idea here is that if there's more than one file found then we need to know which is the one we want to delete, if this if is false it means that `len(refs) == 1`, because if `len(refs) == 0` the first select statement would've failed already.